R/preparedata.R

```diff
@@ -1395,12 +1393,36 @@ transformTable <- function(data,
     # hide.rows.threshold and row.names.to.remove refer to rows AFTER tranposing
     if (isTRUE(transpose))
     {
-        if (length(dim(data)) > 2)
+        if (!is.null(attr(data, "questions")) && length(dim(data)) == 2 && is.null(attr(data, "statistic")))
+        {
+            # 1-dimensional table with multiple statistics
+            is.qtable <- inherits(data, "QTable")
+            old.span <- attr(data, "span", exact = TRUE)
+            new.data <- array(data, dim = c(1, nrow(data), ncol(data)), dimnames = list("", rownames(data), colnames(data)))
+            data <- CopyAttributes(new.data, data)
+            if (is.qtable)
+                class(data) <- c(class(data), "QTable")
+            if (!is.null(old.span))
+                attr(data, "span") <- list(rows = old.span$columns, columns = old.span$rows)
```

Comment:
To get a bit more concrete. This code block has a 1d table with multiple statistic (2d array). It is being 'promoted' to a 3d table with a 1-dimensional slice as the rows. It isn't obvious why this should be done.

Reply:
Yep, this is mainly so that charts with annotations using 1d table with multiple statistics handle the secondary statistics (which might be used for annotations) in the same way as a 2d cross tab with multiple statistics